File: src/main/java/com/shinonometn/re/ssim/repository/CaptureTaskRepository.java
package com.shinonometn.re.ssim.repository;

import com.shinonometn.re.ssim.models.CaptureTask;
import com.shinonometn.re.ssim.models.CaptureTaskDTO;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface CaptureTaskRepository extends CrudRepository<CaptureTask, String> {

    List<CaptureTaskDTO> findAllProjectedBy();

    CaptureTaskDTO findProjectedById(String id);

}

File: src/main/java/com/shinonometn/re/ssim/data/caterpillar/CaptureTaskRepository.kt
package com.shinonometn.re.ssim.data.caterpillar

import org.springframework.data.mongodb.repository.Query
import org.springframework.data.repository.CrudRepository

interface CaptureTaskRepository : CrudRepository<CaptureTask, String> {

    @Query("{}")
    fun findAllProjected(): List<CaptureTaskDTO>

    @Query(value = "{'_id' : ?0}")
    fun findProjectedById(id: String): CaptureTaskDTO

}
